datafusion/datasource-parquet/src/source.rs

```diff
@@ -299,7 +298,7 @@ impl ParquetSource {
     pub fn new(table_schema: impl Into<TableSchema>) -> Self {
         let table_schema = table_schema.into();
         Self {
-            projection: SplitProjection::unprojected(&table_schema),
+            projection: ProjectionExprs::from_schema(table_schema.table_schema()),
```

Comment:
The logic here of creating this upfront even if it will often be replaced is that the most important thing is that each time we create a `ParquetOpener` (once per file) we do the minimum amount of compute. Pre-computing here it and storing it in `ParquetSource` is the easiest way to do that.